daemon/logger/factory/factory.go

```diff
@@ -0,0 +1,57 @@
+package factory
+
+import (
+	"fmt"
+	"sync"
+
+	logger "github.com/docker/docker/daemon/logger"
+)
+
+// Creator is a method that builds a logging driver instance with given context
+type Creator func(LoggingContext) (logger.Logger, error)
+
+// LoggingContext provides enough information for a logging driver to do its function
+type LoggingContext struct {
```

Comment:
Yes, I have. You can use type assertion, I think it is better then make all drivers as bad as worst of them.